app/views/balrog/gate.html.erb

```diff
@@ -1,6 +1,18 @@
 <section>
-  <form action='/balrog/signin' method='POST'>
-    <input autofocus type='password' name='password' placeholder='Password'>
-    <button type='submit'>Login</button>
-  </form>
+  <div>
+    <% if Balrog::Middleware.password_hash || !Balrog::Middleware.omniauth_config %>
```

Comment:
I suggest making these into friendlier booleans within the Balrog code itself, e.g.

`if Balrog.show_password_prompt?` and `if Balrog.omniauth_configured?`

(It can be `Balrog::Middleware.xxx` if necessary)